app/lib/permalink_fu.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module PermalinkFu
+  extend ActiveSupport::Concern
+
+  included do
+    def permalink_must_contain_latin_characters
+      return if permalink.present?
+      errors.add(self.class.permalink_attribute, 'must contain latin characters')
+    end
+  end
+
+  module ClassMethods
+    attr_reader :permalink_options
+    attr_reader :permalink_attribute
+
+    def permalink(attr_name, options = {})
+      @permalink_attribute  = attr_name
+      @permalink_options    = options
+      before_validation :create_unique_permalink
+      validate :permalink_must_contain_latin_characters
+    end
+  end
+
+  private
+
+  def create_unique_permalink
+    base_permalink = permalink.presence || build_permalink_from_attribute
+    count = where_match_permalink_with_conditions(base_permalink).count
+    self.permalink = count.positive? ? "#{base_permalink}-#{count + 1}" : base_permalink
```

Comment:
This is not thread-safe.

Reply:
I know but what can I do about it? I kept the same behaviour we had and I only made it more efficient and clear.
I cannot add a uniqueness validation because as I said in the description of the PR:

> There are actually some forums in the database with the same permalink. I guess that happened before creating this validation of having to be unique.

Reply:
And `forum` does not have a `created_at` to add that validation only for new models 😞

Reply:
I guess it’s not worth the effort of trying to fix. As you said, it’s been like this since forever and AFAIR forum is not something to last very long.